src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1923,6 +1925,7 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Transmit(otRadioFrame &a
         // Waiting for `TransmitDone` event.
         mState        = kStateTransmitting;
         mTxRadioEndUs = otPlatTimeGet() + TX_WAIT_US;
+        mChannel      = mTransmitFrame->mChannel;
```

Comment:
After `SPINEL_PROP_STREAM_RAW` request is sent, the RCP haven't sent the frame because it's doing CMSA backoff. So, the actual channel won't be changed to `mTransmitFrame->mChannel` until CSMA backoff is finished. This will cause a temporary mismatch between `mChannel` and actual RCP channel.
I am not sure if this is going to cause any problem.

Reply:
The safer solution is to avoid caching `mChannel` and always set it on RCP (which has the overhead of extra calls `Set(SPINEL_PROP_PHY_CHAN, ...)` we discussed in https://github.com/openthread/openthread/pull/6837#discussion_r673658912.

I don't think the above situation is an issue.  
- After any backoff delay the radio should perform CA check on the same channel. 
- It is also the same behavior with any radio platform (RCP or not).
- OT core will always issue a `otPlatRadioReceive()` after any frame tx (success or failed) from `UpdateIdleMode()`. 

The `RadioSpinel`/RCP corner case I can think of (which should still work fine) is if RCP hangs/resets in middle of a frame tx and we have the "RCP reset recovery" feature enabled (i.e. `OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0`). 
- In this case the "RCP recovery logic" will set the channel to the frame tx channel.
- But it should be ok, since from OT stack perspective we are in middle of tx (on that channel) and expect a `txdone` callback from which we will call `UpdateIdleMode()` and set the channel to whatever PAN channel is (or basically what OT stack wants it to be).

Reply:
SGTM.